src/vmm/src/devices/virtio/vsock/event_handler.rs

```diff
@@ -182,17 +182,46 @@ where
         let evset = event.event_set();
 
         if self.is_activated() {
-            let mut raise_irq = false;
-            match source {
-                Self::PROCESS_ACTIVATE => self.handle_activate_event(ops),
-                Self::PROCESS_RXQ => raise_irq = self.handle_rxq_event(evset),
-                Self::PROCESS_TXQ => raise_irq = self.handle_txq_event(evset),
-                Self::PROCESS_EVQ => raise_irq = self.handle_evq_event(evset),
-                Self::PROCESS_NOTIFY_BACKEND => raise_irq = self.notify_backend(evset),
-                _ => warn!("Unexpected vsock event received: {:?}", source),
-            }
-            if raise_irq {
-                self.signal_used_queue().unwrap_or_default();
+            let raise_irq = match source {
+                Self::PROCESS_ACTIVATE => {
+                    self.handle_activate_event(ops);
+                    None
+                }
+                Self::PROCESS_RXQ => {
+                    if self.handle_rxq_event(evset) {
+                        Some(Self::PROCESS_RXQ)
+                    } else {
+                        None
+                    }
+                }
+                Self::PROCESS_TXQ => {
+                    if self.handle_txq_event(evset) {
+                        Some(Self::PROCESS_TXQ)
+                    } else {
+                        None
+                    }
+                }
+                Self::PROCESS_EVQ => {
+                    if self.handle_evq_event(evset) {
+                        Some(Self::PROCESS_RXQ)
+                    } else {
+                        None
+                    }
+                }
+                Self::PROCESS_NOTIFY_BACKEND => {
+                    if self.notify_backend(evset) {
+                        Some(Self::PROCESS_NOTIFY_BACKEND)
+                    } else {
+                        None
+                    }
+                }
+                _ => {
+                    warn!("Unexpected vsock event received: {:?}", source);
+                    None
+                }
+            };
+            if let Some(qidx) = raise_irq {
+                self.signal_used_queue(qidx as usize).unwrap_or_default();
```

Comment:
I don't think this call to unwrap_or_default() is needed

Reply:
the `_or_default()` variant was used already, I assume, to ignore errors. I am happy to change this to something like `expect()`. Is that what you mean?

Reply:
Ah, no I was thinking of just doing
```suggestion
                self.signal_used_queue(qidx as usize);
```
but you're right, clippy would complain about the unused results, so it'd have to be
```suggestion
                _ = self.signal_used_queue(qidx as usize).unwrap_or_default();
```
but iirc we do unwrap() on these in other devices, so that should be fine too?

Reply:
Changed it to `expect()`.